### PR TITLE
Oliver parallel

### PR DESCRIPTION
**Pull request by David-Camp**
Hello, 

These changes come from Dan Lipsa at kitware for the Contour Tree work with Gunther Weber.
I have rebase them the latest changes in develop and build them all.
There will also be a pull request for changes in Ascent.

Thanks,
David Camp

### CODE REVIEW

**Comment by David-Camp** [comment]
The build error in the Ubuntu_16 shared and static (maybe more) are because we need "-DVTKM_ENABLE_MPI=TRUE" on the cmake line to build VTK-m with MPI.

File: src/tests/vtkh/CMakeLists.txt [hunk 1]
```diff
@@ -47,13 +47,26 @@
...
 ################################
 
+set(TESTS_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+configure_file(
+  "${CMAKE_CURRENT_SOURCE_DIR}/t_vtk-h_contour_tree.cpp.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/t_vtk-h_contour_tree_par.cpp")
+configure_file(
+  "${CMAKE_CURRENT_SOURCE_DIR}/t_vtk-h_contour_tree.cpp.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/t_vtk-h_contour_tree.cpp")
+
```
Comment (cyrush) [inline src/tests/vtkh/CMakeLists.txt]: avoid configuring the test source, instead we can put the data file in a known place

File: src/vtkh/filters/ContourTree.cpp [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+  int splitDim = -1;
+  for (int i = 1; i < mpi_size; ++i)
+  {
+    for (int j = 0; j < 3; ++j)
+    {
+      if (range[i][2*j] != minRange[j])
+      {
+        blocksPerDim[j] = mpi_size;
+        splitDim = j;
+        goto next; // break with label, Java style
```
Comment (cyrush) [inline src/vtkh/filters/ContourTree.cpp]: remove use of goto

**Comment by cyrush** [comment]
we also want to make sure we update to the newer spack, to follow what ascent does:

https://github.com/Alpine-DAV/ascent/blob/develop/scripts/uberenv/project.json

**Review by mclarsen — APPROVED** [review]

**Pull request merged**